src/Microsoft.ApplicationInsights.AspNetCore/TelemetryInitializers/TelemetryInitializerBase.cs

```diff
@@ -33,15 +33,18 @@ public void Initialize(ITelemetry telemetry)
                 return;
             }
 
-            var request = context.Features.Get<RequestTelemetry>();
-
-            if (request == null)
+            lock (context)
```

Comment:
Yes, but moving it to DI will make us avoid locking. Concurrent dictionary works faster than explicit lock on retrieving the element from regular dictionary.